File: salsel-backend/src/main/java/com/salsel/dto/AddressBookDto.java
package com.salsel.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Data
public class AddressBookDto {
    private Long id;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createdAt;

    private String uniqueId;
    private String name;
    private String contactNumber;
    private String country;
    private String city;
    private String address;
    private String streetName;
    private String district;
    private String refNumber;
    private String userType;
    private Boolean status;
}

File: Salassel-Customer-Ui/src/app/components/addressBook/model/addressBookDto.ts
export interface IAddressBook {
  address?: string;
  city?: string;
  contactNumber?: string;
  country?: string;
  createdAt?: Date;
  district?: string;
  id?: number;
  name?: string;
  refNumber?: string;
  status?: boolean;
  streetName?: string;
  uniqueId?: string;
  userType?: string;
  createdBy?:string
}

export class AddressBook implements IAddressBook {
  constructor(
    public address?: string,
    public city?: string,
    public contactNumber?: string,
    public country?: string,
    public createdAt?: Date,
    public district?: string,
    public id?: number,
    public name?: string,
    public refNumber?: string,
    public status?: boolean,
    public streetName?: string,
    public uniqueId?: string,
    public userType?: string,
    public createdBy?:string
  ) {}
}
